### PR TITLE
[Python] Respect access_token passed into openapi_client.Configuration call

### PR DESCRIPTION
**Pull request by labrown**
Don't overwrite the provided access_token property during configuration call in python client when the spec has a bearer token security schema.

Ran into this issue with the SpaceTraders.io OpenAPI Spec available [here](https://stoplight.io/api/v1/projects/spacetraders/spacetraders/nodes/reference/SpaceTraders.json?fromExportButton=true&snapshotType=http_service&deref=optimizedBundle)

I think this is the correct solution to #13315.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Pinging @spacether @krjakbrjak for attention to this python client change.

### CODE REVIEW

**Comment by spacether** [comment]
How about writing a test? There is no verification that your feature is working.

**Comment by labrown** [comment]
I've spent a couple hours digging around and am not sure where/how to add a test properly.  Can you provide an example Pull Request with a test I could use as a pattern?

**Comment by spacether** [comment]
Sure thing, please add a test in here:
https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python/tests/test_pet_model.py
See any PRs that create or edit files there

File: modules/openapi-generator/src/main/resources/python/configuration.mustache [hunk 1]
```diff
@@ -183,9 +183,6 @@ conf = {{{packageName}}}.Configuration(
         self.password = password
         """Password for HTTP basic authentication
         """
-        self.access_token = access_token
-        """Access token
-        """
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/python/configuration.mustache:188]: for oauth authentication scheme, we still need `access_token`. what about removing line 203-205 instead?

Comment (wing328) [reply modules/openapi-generator/src/main/resources/python/configuration.mustache:188]: Please PM me via Slack so that I can help you with the test.

**Comment by labrown** [comment]
Help me understand the thinking behind the current code.

Line 142 sets `access_token` to `None` as a default value.
Lines 186-188 set `access_token` to the passed in value given to the Configuration call.
Lines 196-200 activate when `#hasOAuthMethods` is true and sets access_token to `None`.  Is there a reason to force it to `None` for OAuth authentication?
Lines 201-207 activate when `hasOAuthMethods` is false and `hasHttpBearerMethods` is true, also setting access_token to `None`.  That seems like the actual bug to me.

Seems like configuration.mustache should generate code that sets `self.access_token`.  It would generate cleaner, more easily understood code.

**Comment by wing328** [comment]
I've merged https://github.com/OpenAPITools/openapi-generator/pull/15802 instead. Please pull the latest master to give it a try.

Happy to reopen this one if needed.

Thanks again for the PR.

**Comment by labrown** [comment]
I'll try it out and see what happens.

**Pull request closed without merging**